docs/ref/models/expressions.txt

```diff
@@ -640,3 +640,28 @@ Let's see how it works::
     Apple: AAPL
     Yahoo: Internet Company
     Django Software Foundation: No Tagline
+
+Adding support for third party backends
+---------------------------------------
+If you are using a backend that uses a different syntax for a certain function,
+you can easily add support for it by monkey patching a new method onto the
+relevant expression class.
+
+Let's say we are using the `Firebird` backend. If `Firebird` would use a
+different syntax for the ``SUM`` function, we could monkey patch a new method
+called ``as_firebird`` onto the ``Sum`` class. We use ``as_firebird`` because
+``connection.vendor`` returns ``firebird`` when using the `Firebird` backend.
+
+In our new method, we can override the default value of ``template`` or
+``function``, or even do more complicated things.
+
+In our example, we could end up with something like this::
+
+    from django.db.models import Sum
+
+    def firebird_sum(self, compiler, connection):
+        template = 'SUM(%(expressions)s)'
+
+        return super(Sum, self).as_sql(compiler, connection, template=template)
```

Comment:
Is the usage of `super()` correct here? The Oracle example earlier in this document and [the code in django-mssql](https://bitbucket.org/Manfre/django-mssql/src/096bd999e4ae032916f567d935b0a647def0997b/sqlserver_ado/expressions.py?at=master&fileviewer=file-view-default) use `self.as_sql()`.